core/src/test/java/integration/kafka/api/PlaintextConsumerDupTest.java

```diff
@@ -0,0 +1,358 @@
+
+    static class SerializerImpl implements Serializer<byte[]> {
+        Serializer<byte[]> serializer = new ByteArraySerializer();
+
+        @Override
+        public byte[] serialize(String topic, byte[] data) {
+            fail("method should not be invoked");
+            return null;
+        }
+
+        @Override
+        public byte[] serialize(String topic, Headers headers, byte[] data) {
+            headers.add("content-type", "application/octet-stream".getBytes());
+            serializer.serialize(topic, data);
+            return data;
+        }
+
+        @Override
+        public void configure(Map<String, ?> configs, boolean isKey) {
+            serializer.configure(configs, isKey);
+        }
+
+        @Override
+        public void close() {
+            serializer.close();
+        }
+    }
+
+    static class DeserializerImpl implements Deserializer<byte[]> {
+        Deserializer<byte[]> deserializer = new ByteArrayDeserializer();
+
+        @Override
+        public byte[] deserialize(String topic, byte[] data) {
+            fail("method should not be invoked");
+            return null;
+        }
+
+        @Override
+        public byte[] deserialize(String topic, Headers headers, byte[] data) {
+            Header header = headers.lastHeader("content-type");
+            if (header != null)
+                assertEquals("application/octet-stream", new String(header.value()));
+            deserializer.deserialize(topic, data);
+            return data;
+        }
+
+        @Override
+        public void configure(Map<String, ?> configs, boolean isKey) {
+            deserializer.configure(configs, isKey);
+        }
+
+        @Override
+        public void close() {
+            deserializer.close();
+        }
+    }
+
+    private void testHeadersSerializeDeserialize(Serializer<byte[]> serializer, Deserializer<byte[]> deserializer) {
+        int numRecords = 1;
+        ProducerRecord<byte[], byte[]> record = new ProducerRecord<>(tp().topic(), tp().partition(), null, "key".getBytes(), "value".getBytes());
+
+        KafkaProducer<byte[], byte[]> producer = createProducer(new ByteArraySerializer(), serializer, new Properties());
+        producer.send(record);
+
+        Consumer<byte[], byte[]> consumer = createConsumer(new ByteArrayDeserializer(), deserializer,
+                new Properties(), JavaConverters.asScalaBuffer(Collections.<String>emptyList()).toList());
+        assertEquals(0, consumer.assignment().size());
+        consumer.assign(Collections.singleton(tp()));
+        assertEquals(1, consumer.assignment().size());
+
+        consumer.seek(tp(), 0);
+        ArrayBuffer<ConsumerRecord<byte[], byte[]>> records = consumeRecords(consumer, numRecords, Integer.MAX_VALUE);
+
+        assertEquals(numRecords, records.size());
+    }
+
+    // Deprecated poll(timeout) not supported for consumer group protocol
+    @Deprecated
+    @Test
+    public void testDeprecatedPollBlocksForAssignment() {
```

Comment:
Are we still supporting deprecated poll for share consumer?

Reply:
In [ShareConsumer.java](https://github.com/confluentinc/kafka/blob/kip-932/clients/src/main/java/org/apache/kafka/clients/consumer/ShareConsumer.java#L58) added by Andrew I do not see poll(long timeout) supported, hence I will remove this test